src/InsertionsClient.Console/ConsoleApp/BuildFilterFactory.cs

```diff
@@ -0,0 +1,246 @@
+                {
+                    Trace.WriteLine($"Failed to parse the rule \"{rule}\": an equals sign should split the rule into two parts.");
+                    return false;
+                }
+
+                string propertyName = sides[0];
+                string regularExp = sides[1];
+                Regex? regex = null;
+
+                try
+                {
+                    regex = new Regex(regularExp, RegexOptions.Compiled);
+                }
+                catch (Exception e)
+                {
+                    Trace.WriteLine("Failed to parse the regular expression: " + regularExp + ": " + e.ToString());
+                    return false;
+                }
+
+                if (!TryGetBuildSelectorFromPropertyName(propertyName, regex, out Predicate<Build>? buildSelector)
+                    || buildSelector == null)
+                {
+                    Trace.WriteLine("Build selector couldn't be constructed from the given property name: " + propertyName);
+                    return false;
+                }
+
+                MethodCallExpression ruleExpression = Expression.Call(Expression.Constant(buildSelector.Target), buildSelector.Method, buildParameter);
+
+                if (rulesetExpression == null)
+                {
+                    // We haven't put any code into the expression of this ruleset. This will be the first.
+                    rulesetExpression = ruleExpression;
+                }
+                else
+                {
+                    // This ruleset already contains some code from the previous rules. Combine them with "And".
+                    rulesetExpression = Expression.AndAlso(rulesetExpression, ruleExpression);
+                }
+            }
+
+            Trace.WriteLine("Failed to generate an expression from the given ruleset: " + ruleset);
+            return rulesetExpression != null;
+        }
+
+        /// <summary>
+        /// Attempts to create a predicate with the given property name and the regular expression.
+        /// The predicate (1) takes a build as argument,
+        /// (2) accesses the property with the given name,
+        /// (3) tries to match it with the provided regular expression,
+        /// and (4) returns the result of the match.
+        /// </summary>
+        /// <param name="propertyName">Name of the property in the <see cref="Build"/> instance to be inputted to regex match.</param>
+        /// <param name="regex">Regular expression to test the property with.</param>
+        /// <param name="buildSelector">The generated predicate that can be used to check if a build succeeds matching its
+        /// property with the regular expression.</param>
+        /// <returns>True if creating a predicate was successful. False, otherwise.</returns>
+        private static bool TryGetBuildSelectorFromPropertyName(string propertyName, Regex regex, out Predicate<Build>? buildSelector)
+        {
+            switch (propertyName)
```

Comment:
not actionable: for your benefit, take a look at the newer switch statement format.  In my opinion, more elegant.

Reply:
Thanks. I tried to use it here, but I couldn't find a way to print an error in the default case.